android/java/org/chromium/chrome/browser/toolbar/adaptive/BraveNewsButtonController.java

```diff
@@ -0,0 +1,71 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.toolbar.adaptive;
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.graphics.drawable.Drawable;
+import android.view.View;
+
+import org.chromium.base.supplier.ObservableSupplier;
+import org.chromium.build.annotations.NullMarked;
+import org.chromium.build.annotations.Nullable;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.ActivityTabProvider;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.profiles.Profile;
+import org.chromium.chrome.browser.tab.Tab;
+import org.chromium.chrome.browser.toolbar.optional_button.BaseButtonDataProvider;
+import org.chromium.ui.modaldialog.ModalDialogManager;
+
+/** Handles displaying Brave News button on toolbar. */
+@NullMarked
+public class BraveNewsButtonController extends BaseButtonDataProvider {
+    private final Context mContext;
+    private final ObservableSupplier<Profile> mProfileSupplier;
+
+    public BraveNewsButtonController(
+            Context context,
+            Drawable buttonDrawable,
+            ActivityTabProvider tabProvider,
+            ObservableSupplier<Profile> profileSupplier,
+            ModalDialogManager modalDialogManager) {
+        super(
+                tabProvider,
+                modalDialogManager,
+                buttonDrawable,
+                context.getString(R.string.brave_news_title),
+                /* actionChipLabelResId= */ Resources.ID_NULL,
+                /* supportsTinting= */ true,
+                /* iphCommandBuilder= */ null,
+                AdaptiveToolbarButtonVariant.NEWS,
+                /* tooltipTextResId= */ R.string.brave_news_title);
+
+        mContext = context;
+        mProfileSupplier = profileSupplier;
+    }
+
+    @Override
+    public void onClick(View view) {
+        Profile profile = mProfileSupplier.get();
+        if (profile == null) return;
+
+        assert mContext instanceof BraveActivity : "Context is not an BraveActivity";
+
+        // Open Brave News Settings
+        if (mContext instanceof BraveActivity braveActivity) {
+            braveActivity.openBraveNewsSettings();
```

Comment:
should it open news settings or the news itself at the new tab?

Reply:
That's a good catch, currently it does the same action as menu item does (which is opening settings and doesn't look really helpful). @timchilds what is actual desired behavior for `Brave News` shortcut?